cmd/cloud-controller-manager/app/dynamic/file_watcher.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamic
+
+import (
+	"os"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+
+	"k8s.io/klog/v2"
+)
+
+func RunFileWatcherOrDie(path string) chan struct{} {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		klog.Errorf("RunFileWatcherOrDie: failed to initialize file watcher: %s", err)
+		os.Exit(1)
+	}
+
+	startWatchingOrDie(watcher, path)
+
+	updateChan := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					klog.Error("RunFileWatcherOrDie: events channel closed unexpectedly")
+					_ = watcher.Close()
+					os.Exit(1)
+				}
+
+				klog.Infof("RunFileWatcherOrDie: found file update event: %v", event)
+				updateChan <- struct{}{}
+
+				if strings.EqualFold(event.Name, path) && event.Op&fsnotify.Remove == fsnotify.Remove {
```

Comment:
it's better to have a retry (e.g. try 5 times with 1 second interval) to avoid rapid die on the config file changes manually (for example, cluster operator may want to delete the file and copy it to the same filename again)

Reply:
done